### PR TITLE
Add documentation for Apollo Federation support

### PR DESCRIPTION
**Pull request by rafalp**
Adds documentation for changes introduced in https://github.com/mirumee/saleor/pull/8273 but also adds bunch of extra knowledge that should be useful to folk seeking to federate their GraphQL services with Saleor.

### CODE REVIEW

File: docs/developer/apollo-federation.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+| `ProductMedia`    | `id`          |                   |
+| `ProductType`     | `id`          |                   |
+| `ProductVariant`  | `id channel`  |                   |
+| `User`            | `id`          | `email`           |
+
+## Known issues and limitations
+
+Apollo Federation doesn't handle the scenario when federated server didn't return the requested object [too well](https://github.com/apollographql/federation/issues/376). GraphQL gateway will report an error when Saleor fails to return requested entity, either because it was deleted, request is missing authorization required to retrieve it or it's ID was incorrect.
+
+Rely on webhooks to update or clear relations stored in your service's database when objects are deleted in the Saleor.
```
Comment (maarcingebala) [inline docs/developer/apollo-federation.mdx:31]:
```suggestion
Rely on webhooks to update or clear relations stored in your service's database when objects are deleted in Saleor.
```

File: docs/developer/running-saleor/configuration.mdx [hunk 2]
```diff
@@ -173,6 +173,10 @@ Controls whether registering new customer accounts should require email confirma
 
 Defaults to `True`.
 
+### `FEDERATED_QUERY_MAX_ENTITIES`
+
+Sets upper limit on how many entities may be requested by Apollo Federation's gateway for single GraphQL Query. Defaults to `100`. Set to `0` to remove this limit.
```
Comment (maarcingebala) [inline docs/developer/running-saleor/configuration.mdx:178]:
```suggestion
Sets upper limit on how many entities may be requested by Apollo Federation's gateway for a single GraphQL query. Defaults to `100`. Set to `0` to remove this limit.
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

**Comment by rafalp** [comment]
Moved federation docs under dev/extending as suggested by @krzysztofwolski

**Review by maarcingebala — APPROVED** [review]

**Comment by maarcingebala** [comment]
@krzysztofwolski Can you re-review this one?

**Review by krzysztofwolski — APPROVED** [review]

**Comment by krzysztofwolski** [comment]
LGTM!

**Pull request merged**